engine/consensus/matching/core_test.go

```diff
@@ -556,9 +618,57 @@ func (ms *MatchingSuite) TestRequestPendingReceipts() {
+	ms.ReceiptsDB.On("ByBlockIDAllExecutionReceipts", mock.Anything).Return(nil, nil)
+
+	_, _, err := ms.matching.requestPendingReceipts()
+	ms.Require().NoError(err, "should request results for pending blocks")
+	ms.requester.AssertExpectations(ms.T()) // asserts that requester.Query(<blockID>, filter.Any) was called
+}
+
+// TestRequestSecondPendingReceipt verifies that a second receipt is re-requested
+// Situation A:
+//  * we have _once_ receipt for an unsealed finalized block in storage
+//  * Expected: Method Core.requestPendingReceipts() should re-request a second receipt
+// Situation B:
+//  * we have _two_ receipts for an unsealed finalized block storage
+//  * Expected: Method Core.requestPendingReceipts() should _not_ request another receipt
+//
+// TODO: this test is temporarily requires as long as matching.Core requires _two_ receipts from different ENs to seal
+func (ms *MatchingSuite) TestRequestSecondPendingReceipt() {
+	ms.matching.sealingThreshold = 0 // request receipts for all unsealed finalized blocks
+
+	result := unittest.ExecutionResultFixture(unittest.WithBlock(ms.LatestFinalizedBlock))
+
+	// add an incorporatedResult for finalized block
+	// TODO: update WithIncorporatedBlockID once we move to sealing Phase 3
+	incRes := unittest.IncorporatedResult.Fixture(
+		unittest.IncorporatedResult.WithResult(result),
+		unittest.IncorporatedResult.WithIncorporatedBlockID(ms.LatestFinalizedBlock.ID()),
+	)
+	ms.PendingResults[incRes.ID()] = incRes
+
+	// make receipts:
+	receipt1 := unittest.ExecutionReceiptFixture(unittest.WithResult(result))
+	receipt2 := unittest.ExecutionReceiptFixture(unittest.WithResult(result))
+
+	// receipts from storage are potentially added to receipts mempool and incorporated results mempool
+	ms.ReceiptsPL.On("AddReceipt", receipt1, ms.LatestFinalizedBlock.Header).Return(false, nil).Maybe()
+	ms.ReceiptsPL.On("AddReceipt", receipt2, ms.LatestFinalizedBlock.Header).Return(false, nil).Maybe()
+	ms.ResultsPL.On("Add", incRes).Return(false, nil).Maybe()
+
+	// Situation A: we have _once_ receipt for an unsealed finalized block in storage
+	ms.ReceiptsDB.On("ByBlockIDAllExecutionReceipts", ms.LatestFinalizedBlock.ID()).Return([]*flow.ExecutionReceipt{receipt1}, nil).Once()
+	ms.requester.On("Query", ms.LatestFinalizedBlock.ID(), mock.Anything).Return().Once() // Core should trigger requester to re-request a second receipt
 	_, _, err := ms.matching.requestPendingReceipts()
 	ms.Require().NoError(err, "should request results for pending blocks")
-	ms.requester.AssertExpectations(ms.T()) // asserts that requester.EntityByID(<blockID>, filter.Any) was called
+	ms.requester.AssertExpectations(ms.T()) // asserts that requester.Query(<blockID>, filter.Any) was called
+
+	// Situation B: we have _two_ receipts for an unsealed finalized block storage
+	ms.ReceiptsDB.On("ByBlockIDAllExecutionReceipts", ms.LatestFinalizedBlock.ID()).Return([]*flow.ExecutionReceipt{receipt1, receipt2}, nil).Once()
```

Comment:
we don't have `ByBlockIDAllExecutionReceipts` anymore, is this test broken?

Reply:
thanks. Fixed tests